webhookClient/periodicRegisterer_test.go

```diff
@@ -17,4 +17,81 @@
 
 package webhookClient
 
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/xmidt-org/webpa-common/xmetrics/xmetricstest"
+	webhook "github.com/xmidt-org/wrp-listener"
+)
+
 // TODO: add unit tests
+
+func TestNewPeriodicRegisterer(t *testing.T) {
+	mockAcquirer := new(MockAcquirer)
+	mockSecretGetter := new(mockSecretGetter)
+	registry := xmetricstest.NewProvider(nil, Metrics)
+	m := NewMeasures(registry)
+
+	basicRegisterer := BasicRegisterer{
+		acquirer:        mockAcquirer,
+		secretGetter:    mockSecretGetter,
+		requestTemplate: webhook.W{},
+		registrationURL: "random string",
+	}
+
+	logger := log.NewNopLogger()
+	validInterval, _ := time.ParseDuration("10s")
+
+	tests := []struct {
+		description        string
+		registerer         Registerer
+		interval           time.Duration
+		logger             log.Logger
+		expectedRegisterer *PeriodicRegisterer
+		expectedErr        error
+	}{
+		{
+			description: "Success",
+			registerer:  &basicRegisterer,
+			interval:    validInterval,
+			logger:      logger,
+			expectedRegisterer: &PeriodicRegisterer{
+				registerer:           &basicRegisterer,
+				registrationInterval: validInterval,
+				logger:               logger,
+				measures:             m,
+			},
+			expectedErr: nil,
+		},
+		{
```

Comment:
These tests are great! Can you add a test case that verifies the default logger is added when a nil logger is given? I would call it something like `Success with Default Logger`, but whatever you think is good.